skt/executable.py

```diff
@@ -397,125 +397,266 @@ def setup_parser():
-                        help="Path to dir to store junit results in")
-    parser.add_argument("-v", "--verbose", help="Increase verbosity level",
-                        action="count", default=0)
-    parser.add_argument("--rc", help="Path to rc file", default=DEFAULTRC)
+    description_text = """
+skt is a tool for automatically fetching, building, and testing kernel patches
+published on Patchwork instances.
+
+Several subcommands are available:
+
+    all     - run merge, build, publish, run, report, cleanup in order
+    bisect  - bisect the git history to find the offending commit
+    build   - build the kernel with the specified configuration and package it
+              into a tarball
+    cleanup - remove build information, kernel tarball, state information
+              (if --state specified), working directory (if --wipe specified)
+    merge   - apply a patch from an instance of patchwork to a specified
+              reference in a git repository
+    publish - copy the kernel tarball, configuration, and build information to
+              a specified location (cp and scp publishers supported)
+    report  - report build/test results using a reporter (email and stdout
+              reporters supported)
+    run     - run tests on a built kernel using a runner (beaker is the only
+              runner currently supported)
+"""
+
+    parser = argparse.ArgumentParser(
+        add_help=False,
+        description=description_text,
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    # Global arguments for all skt operations
+    parser.add_argument(
+        "-d", "--workdir",
+        type=str,
+        help="Path to work dir"
+    )
+    parser.add_argument(
+        "-w", "--wipe",
+        help="Clean build (make mrproper before building), remove workdir "
+             "when finished",
+        action="store_true",
+        default=False
+    )
+    parser.add_argument(
+        "--junit",
+        help="Path to dir to store junit results in"
+    )
+    parser.add_argument(
+        "-v", "--verbose",
+        help="Increase verbosity level",
+        action="count",
+        default=0
+    )
+    parser.add_argument(
+        "--rc",
+        help="Path to rc file",
+        default=DEFAULTRC
+    )
     # FIXME Storing state in config file can break the whole system in case
     #       state saving aborts. It's better to save state separately.
     #       It also breaks separation of concerns, as in principle skt doesn't
     #       need to modify its own configuration otherwise.
-    parser.add_argument("--state", help="Save/read state from 'state' section "
-                        "of rc file", action="store_true", default=False)
+    parser.add_argument(
+        "--state",
+        help="Save/read state from 'state' section of rc file",
+        action="store_true",
+        default=False
+    )
+    parser.add_argument(
+        "-h", "--help",
+        help="Merge sub-command help",
```

Comment:
Hm, shouldn't this be a generic help? I never reads actual messages, unless I start debugging things...